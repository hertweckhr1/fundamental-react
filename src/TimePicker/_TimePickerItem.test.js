import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import TimePickerItem from './_TimePickerItem';

describe('<TimePickerItem />', () => {
    const defaultTimePickerItem = <TimePickerItem id='myTime' />;
    const twelveHourTime = <TimePickerItem format12Hours />;
    const showHour = <TimePickerItem format12Hours showHour />;
    const showMinute = <TimePickerItem format12Hours showMinute />;
    const showSecond = <TimePickerItem format12Hours showSecond />;
    const showHourMinute = (
        <TimePickerItem format12Hours={false} showHour={false}
            showMinute />
    );
    const showMinuteSecond = (
        <TimePickerItem
            format12Hours
            showHour={false}
            showMinute
            showSecond />
    );
    const showHourSecond = (
        <TimePickerItem
            format12Hours
            showHour
            showMinute={false}
            showSecond />
    );
    const noSecondTime = <TimePickerItem showSecond={false} />;
    const disabledTime = <TimePickerItem disabled />;
    test('create time picker item', () => {
        // default time picker
        let component = renderer.create(defaultTimePickerItem);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // twelve hour time picker
        component = renderer.create(twelveHourTime);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // no seconds time picker
        component = renderer.create(noSecondTime);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // disabled time picker
        component = renderer.create(disabledTime);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component = renderer.create(showHour);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component = renderer.create(showSecond);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component = renderer.create(showMinute);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component = renderer.create(showHourMinute);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component = renderer.create(showMinuteSecond);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component = renderer.create(showHourSecond);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the TimePickerItem component\'s input element', () => {
            const element = mount(<TimePickerItem id='id' inputProps={{ 'data-sample': 'Sample' }} />);

            expect(
                element.find('#id-input').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        test('should allow props to be spread to the TimePickerItem component\'s button element', () => {
            const element = mount(<TimePickerItem buttonProps={{ 'data-sample': 'Sample' }} id='id' />);

            expect(
                element.find('#id-button').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
