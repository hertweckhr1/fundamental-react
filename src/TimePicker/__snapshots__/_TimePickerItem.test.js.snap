// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimePickerItem /> create time picker item 1`] = `
<div
  className="fd-popover__control"
>
  <div
    className="fd-input-group fd-input-group--after"
  >
    <input
      className="fd-input"
      id="myTime-input"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
    <span
      className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button "
    >
      <button
        aria-controls="rthHR811"
        aria-expanded="false"
        aria-haspopup="true"
        className="fd-button--light fd-button--compact sap-icon--fob-watch fd-popover__control"
        id="myTime-button"
      />
    </span>
  </div>
</div>
`;

exports[`<TimePickerItem /> create time picker item 2`] = `
<div
  className="fd-popover__control"
>
  <div
    className="fd-input-group fd-input-group--after"
  >
    <input
      className="fd-input"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
    <span
      className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button "
    >
      <button
        aria-controls="rthHR811"
        aria-expanded="false"
        aria-haspopup="true"
        className="fd-button--light fd-button--compact sap-icon--fob-watch fd-popover__control"
      />
    </span>
  </div>
</div>
`;

exports[`<TimePickerItem /> create time picker item 3`] = `
<div
  className="fd-popover__control"
>
  <div
    className="fd-input-group fd-input-group--after"
  >
    <input
      className="fd-input"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
    <span
      className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button "
    >
      <button
        aria-controls="rthHR811"
        aria-expanded="false"
        aria-haspopup="true"
        className="fd-button--light fd-button--compact sap-icon--fob-watch fd-popover__control"
      />
    </span>
  </div>
</div>
`;

exports[`<TimePickerItem /> create time picker item 4`] = `
<div
  className="fd-popover__control"
>
  <div
    className="fd-input-group fd-input-group--after"
  >
    <input
      className="fd-input"
      onBlur={[Function]}
      onChange={[Function]}
      readOnly={true}
      type="text"
    />
    <span
      className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button "
    >
      <button
        aria-controls="rthHR811"
        aria-expanded="false"
        aria-haspopup="true"
        className="fd-button--light fd-button--compact sap-icon--fob-watch fd-popover__control"
        disabled={true}
      />
    </span>
  </div>
</div>
`;

exports[`<TimePickerItem /> create time picker item 5`] = `
<div
  className="fd-popover__control"
>
  <div
    className="fd-input-group fd-input-group--after"
  >
    <input
      className="fd-input"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
    <span
      className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button "
    >
      <button
        aria-controls="rthHR811"
        aria-expanded="false"
        aria-haspopup="true"
        className="fd-button--light fd-button--compact sap-icon--fob-watch fd-popover__control"
      />
    </span>
  </div>
</div>
`;

exports[`<TimePickerItem /> create time picker item 6`] = `
<div
  className="fd-popover__control"
>
  <div
    className="fd-input-group fd-input-group--after"
  >
    <input
      className="fd-input"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
    <span
      className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button "
    >
      <button
        aria-controls="rthHR811"
        aria-expanded="false"
        aria-haspopup="true"
        className="fd-button--light fd-button--compact sap-icon--fob-watch fd-popover__control"
      />
    </span>
  </div>
</div>
`;

exports[`<TimePickerItem /> create time picker item 7`] = `
<div
  className="fd-popover__control"
>
  <div
    className="fd-input-group fd-input-group--after"
  >
    <input
      className="fd-input"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
    <span
      className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button "
    >
      <button
        aria-controls="rthHR811"
        aria-expanded="false"
        aria-haspopup="true"
        className="fd-button--light fd-button--compact sap-icon--fob-watch fd-popover__control"
      />
    </span>
  </div>
</div>
`;

exports[`<TimePickerItem /> create time picker item 8`] = `
<div
  className="fd-popover__control"
>
  <div
    className="fd-input-group fd-input-group--after"
  >
    <input
      className="fd-input"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
    <span
      className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button "
    >
      <button
        aria-controls="rthHR811"
        aria-expanded="false"
        aria-haspopup="true"
        className="fd-button--light fd-button--compact sap-icon--fob-watch fd-popover__control"
      />
    </span>
  </div>
</div>
`;

exports[`<TimePickerItem /> create time picker item 9`] = `
<div
  className="fd-popover__control"
>
  <div
    className="fd-input-group fd-input-group--after"
  >
    <input
      className="fd-input"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
    <span
      className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button "
    >
      <button
        aria-controls="rthHR811"
        aria-expanded="false"
        aria-haspopup="true"
        className="fd-button--light fd-button--compact sap-icon--fob-watch fd-popover__control"
      />
    </span>
  </div>
</div>
`;

exports[`<TimePickerItem /> create time picker item 10`] = `
<div
  className="fd-popover__control"
>
  <div
    className="fd-input-group fd-input-group--after"
  >
    <input
      className="fd-input"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
    <span
      className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button "
    >
      <button
        aria-controls="rthHR811"
        aria-expanded="false"
        aria-haspopup="true"
        className="fd-button--light fd-button--compact sap-icon--fob-watch fd-popover__control"
      />
    </span>
  </div>
</div>
`;
